source/extensions/filters/common/rbac/engine_impl.cc

```diff
@@ -1,23 +1,40 @@
 #include "source/extensions/filters/common/rbac/engine_impl.h"
 
 #include "envoy/config/rbac/v3/rbac.pb.h"
+#include "envoy/extensions/rbac/custom_library_config/v3/custom_library.pb.h"
+#include "envoy/protobuf/message_validator.h"
 
+#include "source/common/config/utility.h"
 #include "source/common/http/header_map_impl.h"
+#include "source/extensions/filters/common/expr/library/custom_library.h"
 
 namespace Envoy {
 namespace Extensions {
 namespace Filters {
 namespace Common {
 namespace RBAC {
 
+using BaseCustomLibraryFactory =
+    Envoy::Extensions::Filters::Common::Expr::Library::BaseCustomLibraryFactory;
+
 RoleBasedAccessControlEngineImpl::RoleBasedAccessControlEngineImpl(
     const envoy::config::rbac::v3::RBAC& rules,
     ProtobufMessage::ValidationVisitor& validation_visitor, const EnforcementMode mode)
     : action_(rules.action()), mode_(mode) {
   // guard expression builder by presence of a condition in policies
+
+  if (rules.has_custom_library_config()) {
+    auto& factory = Envoy::Config::Utility::getAndCheckFactory<BaseCustomLibraryFactory>(
+        rules.custom_library_config());
+    custom_library_ = factory.createLibrary(rules.custom_library_config(), validation_visitor);
+  } else {
+    custom_library_ = nullptr;
+  }
+
   for (const auto& policy : rules.policies()) {
     if (policy.second.has_condition()) {
-      builder_ = Expr::createBuilder(&constant_arena_);
+      builder_ =
+          Expr::createBuilder(&constant_arena_, custom_library_ ? custom_library_.get() : nullptr);
```

Comment:
I believe this is equivalent to:
```suggestion
          Expr::createBuilder(&constant_arena_, custom_library_.get());
```

Reply:
ok